sycl/test/basic_tests/known_identity.cpp

```diff
@@ -39,15 +45,52 @@ template <typename T> void checkIntKnownIdentity() {
   static_assert(known_identity<std::bit_and<>, T>::value == Ones);
   static_assert(known_identity<std::bit_and<T>, T>::value == Ones);
 
+  static_assert(has_known_identity<sycl::bit_and<>, T>::value);
+  static_assert(has_known_identity<sycl::bit_and<T>, T>::value);
+  static_assert(known_identity<sycl::bit_and<>, T>::value == Ones);
+  static_assert(known_identity<sycl::bit_and<T>, T>::value == Ones);
+
   static_assert(has_known_identity<std::bit_or<>, T>::value);
   static_assert(has_known_identity<std::bit_or<T>, T>::value);
   static_assert(known_identity<std::bit_or<>, T>::value == 0);
   static_assert(known_identity<std::bit_or<T>, T>::value == 0);
 
+  static_assert(has_known_identity<sycl::bit_or<>, T>::value);
+  static_assert(has_known_identity<sycl::bit_or<T>, T>::value);
+  static_assert(known_identity<sycl::bit_or<>, T>::value == 0);
+  static_assert(known_identity<sycl::bit_or<T>, T>::value == 0);
+
   static_assert(has_known_identity<std::bit_xor<>, T>::value);
   static_assert(has_known_identity<std::bit_xor<T>, T>::value);
   static_assert(known_identity<std::bit_xor<>, T>::value == 0);
   static_assert(known_identity<std::bit_xor<T>, T>::value == 0);
+
+  static_assert(has_known_identity<sycl::bit_xor<>, T>::value);
+  static_assert(has_known_identity<sycl::bit_xor<T>, T>::value);
+  static_assert(known_identity<sycl::bit_xor<>, T>::value == 0);
+  static_assert(known_identity<sycl::bit_xor<T>, T>::value == 0);
+}
+
+template <typename T> void checkBoolKnownIdentity() {
```

Comment:
Should it also be possible to use `bit_and`, `bit_or`, `bit_xor` with `bool`?  I think these operations are well defined.

Reply:
Interesting question, but I afraid it may be implementation defined. At least this doc does not mention bool type: https://en.cppreference.com/w/cpp/utility/functional/bit_and

Reply:
@Pennycook - This PR passed testing, it fixes an obvious miss. bit_and for bool is a separate and less obvious (imo). It can be supported by another PR. If you don't mind I'll merge the current PR as is.

Reply:
Sorry for the delayed response! Yes, I agree that this can be merged, and we can revisit the question of which type/operator combinations are supported later.

I think things like `bit_and` _do_ work for `bool`, but possibly because of weird integer conversion rules in C++.  It's not immediately obvious to me which combinations the SYCL specification expects to work; I'll try to get some more clarity on this point, and we can also wait to see if any users actually complain about not being able to use "weird" combinations.